lib/charms/mysql/v0/mysql.py

```diff
@@ -1599,6 +1599,24 @@ def initialize_juju_units_operations_table(self) -> None:
             )
             raise MySQLInitializeJujuOperationsTableError(e.message)
 
+    def rescan_cluster_if_instance_already_in_cluster(
```

Comment:
I would prefer to modify the existing method `in_instance_in_cluster` to have the `from_instance` parameter, and test for member states there, so we can ditch this new method.

Then call both methods (is_instance_in_cluster and rescan) within the charm code, so the flow is clear there